client/packages/system/src/Patient/Components/utils.ts

```diff
@@ -1,27 +1,29 @@
 import { useEffect, useState } from 'react';
-import { usePatient } from '../api';
-import { useDebouncedValueCallback } from '@common/hooks';
+import { ProgramPatientRowFragment, usePatient } from '../api';
+import { useDebouncedValue } from '@common/hooks';
 
+const shouldSearch = (text: string) => text.length > 0;
 export const useSearchPatient = () => {
   const [searchText, setSearchText] = useState('');
   const { mutate, isLoading, data, isSuccess } = usePatient.utils.search();
 
-  const debounced = useDebouncedValueCallback(
-    value => mutate({ nameOrCode: value }),
-    [searchText],
-    500
-  );
```

Comment:
Not sure what caused it but for some reason it didn't had the current value of searchText when calling `mutate()`...